general/project/repoutils.go

```diff
@@ -0,0 +1,137 @@
+	NugetVirtualDefaultName  = "default-nuget-virtual"
+)
+
+var RepoDefaultName = map[coreutils.Technology]map[string]string{
+	coreutils.Maven: {
+		Local:     MavenLocalDefaultName,
+		Remote:    MavenRemoteDefaultName,
+		RemoteUrl: MavenRemoteDefaultUrl,
+		Virtual:   MavenVirtualDefaultName,
+	},
+	coreutils.Gradle: {
+		Local:     GradleLocalDefaultName,
+		Remote:    GradleRemoteDefaultName,
+		RemoteUrl: GradleRemoteDefaultUrl,
+		Virtual:   GradleVirtualDefaultName,
+	},
+	coreutils.Npm: {
+		Local:     NpmLocalDefaultName,
+		Remote:    NpmRemoteDefaultName,
+		RemoteUrl: NpmRemoteDefaultUrl,
+		Virtual:   NpmVirtualDefaultName,
+	},
+	coreutils.Go: {
+		Local:     GoLocalDefaultName,
+		Remote:    GoRemoteDefaultName,
+		RemoteUrl: GoRemoteDefaultUrl,
+		Virtual:   GoVirtualDefaultName,
+	},
+	coreutils.Pypi: {
+		Local:     PypiLocalDefaultName,
+		Remote:    PypiRemoteDefaultName,
+		RemoteUrl: PypiRemoteDefaultUrl,
+		Virtual:   PypiVirtualDefaultName,
+	},
+}
+
+func CreateDefaultLocalRepo(technologyType coreutils.Technology, serverId string) error {
+	servicesManager, err := getServiceManager(serverId)
+	if err != nil {
+		return err
+	}
+	params := services.NewLocalRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Local]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateLocalRepositoryWithParams(params)
+}
+
+func CreateDefaultRemoteRepo(technologyType coreutils.Technology, serverId string) error {
+	servicesManager, err := getServiceManager(serverId)
+	if err != nil {
+		return err
+	}
+	params := services.NewRemoteRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Remote]
+	params.Url = RepoDefaultName[technologyType][RemoteUrl]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateRemoteRepositoryWithParams(params)
+}
+
+func CreateDefaultVirtualRepo(technologyType coreutils.Technology, serverId string) error {
+	servicesManager, err := getServiceManager(serverId)
+	if err != nil {
+		return err
+	}
+	params := services.NewVirtualRepositoryBaseParams()
+	params.PackageType = string(technologyType)
+	params.Key = RepoDefaultName[technologyType][Virtual]
+	params.Repositories = []string{RepoDefaultName[technologyType][Local], RepoDefaultName[technologyType][Remote]}
+	params.DefaultDeploymentRepo = RepoDefaultName[technologyType][Local]
+	if repoExists(servicesManager, params.Key) {
+		return nil
+	}
+	return servicesManager.CreateVirtualRepositoryWithParams(params)
+}
+
+func getServiceManager(serverId string) (artifactory.ArtifactoryServicesManager, error) {
+	serviceDetails, err := config.GetSpecificConfig(serverId, true, false)
+	if err != nil {
+		return nil, err
+	}
+	return rtUtils.CreateServiceManager(serviceDetails, -1, false)
+
+}
+
+// Check if default repository is allready exists
```

Comment:
// Check if default repository is allready exists
-->
// Check if default repository already exists